File: examples/javascript/singleton.js
class Singleton {
    constructor() {
        if (Singleton._instance)
            throw new Error("instantiated already");

        Singleton._instance = this;
    }
    //some code
    getData() {
        return this.Data;
    }

    setData(data) {
        this.Data = data;
    }
}

let c = new Singleton();
console.log(c.setData(12));
console.log(c.getData());



const test = () => {
    this.c = 22;
    this.a = function () {
        console.log("ok");
    }
}

File: examples/python/Singleton.py
class Singleton:
    instance = None
    def __init__(self):
       self._data = 12 
       if not Singleton.instance:
           Singleton.instance = self
    
    @staticmethod
    def getInstance():
        if Singleton.instance is None:
            Singleton()
        else:
            raise Exception("Singleton class already defined")
        return Singleton.instance

    @property
    def Data(self):
        return self._data
    
    @Data.setter
    def Data(self,a):
        self._data = a



print(Singleton.getInstance().Data)
Singleton.getInstance().Data = 16
print(Singleton.getInstance().Data)
